ui/src/app/views/quick-tour-modal/component.tsx

```diff
@@ -0,0 +1,222 @@
+    },
+    {
+      title: 'Cohorts',
+      content: <div>A “Cohort” is a group of participants you are interested in researching.
+        The Cohort Builder allows you to create and review cohorts and annotate
+        participants in your study group.
+        <br/><br/>
+        For example, you can build a Cohort called “diabetes cases,” to include people
+        who have been diagnosed with type II diabetes, using a combination of billing codes and
+        laboratory values. You can also have a “controls” Cohort. Once you build your cohorts,
+        you can go through and manually review the records for each participant and decide if
+        you want to include or exclude them from your Cohort and make specific
+        annotations/notes to each record.</div>,
+      image: '/assets/images/cohorts.png'
+    },
+    {
+      title: 'Concepts',
+      content: <div>Concepts describe information in a patient’s medical record, such as a condition
+          they have, a  prescription they are taking or their physical measurements. In the Workbench
+          we refer to subject areas such as conditions, drugs, measurements etc. as “domains.” You can
+          search for and save collections of concepts from a particular domain as a “Concept Set.”
+          <br/><br/>
+          For example, if you want to select height, weight and blood pressure information
+          (concepts) from your “diabetes cases” Cohort, you can search for the 3 concepts
+          from the “Measurements” domain and call it “biometrics” Concept Set. You can then
+          use Notebooks to extract that information from your cohort.</div>,
+      image: '/assets/images/concepts.png'
+    },
+    {
+      title: 'Notebooks',
+      content: <div>A Notebook is a computational environment where you can analyze data with basic
+          programming knowledge in R or Python. Several template Notebooks and resources
+          are available within your Workspace that will guide you how to import your
+          Cohort(s) and Concept Set(s) into the Notebook and can assist with basic analyses.
+          <br/><br/>
+         For example, you can launch a Notebook
+          to import your “diabetes cases” Cohort and then select your “biometrics” Concept Set, to
+          get biometrics data for the participants in your Cohort. You can then analyze the data to
+          study correlation between hypertension and diabetes.</div>,
+      image: '/assets/images/notebooks.png'
+    }];
+
+class QuickTourReact extends React.Component<any, any> {
+  state: {selected: number, fullImage: boolean, numPanels: number};
+  props: {learning: boolean, closeFunction: Function};
+
+  checkImg = '/assets/images/check.svg';
+  expandIcon = '/assets/icons/expand.svg';
+
+  constructor(props: Object) {
+    super(props);
+    this.state = {selected: 0, fullImage: false, numPanels: 5};
+  }
+
+  previous(): void {
+    this.setState({selected: this.state.selected - 1});
```

Comment:
Based on my limited understanding of React, when you reference `state` you should instead use the callback invocation form as this update may be applied asynchronously:

```
this.setState(state => {
  return {selected: state.selected - 1};
});
```

Reply:
While not strictly necessary - react will shallow merge state when called directly - I think the functional form is [preferred](https://reactjs.org/docs/react-component.html#setstate). 

To take this a step further, if you extracted the functional call into a separate function, you can gain some unit-testability.
```
function decrementSelected(state, props) { // setState takes two args, current state and props
  return {selected: state.selected - 1}
}
... 
this.setState(decrementSelected)
```

Reply:
The function form is indeed preferred, especially in the case of multiple simultaneous updates. In practice, I find that scenario doesn't happen much, so the object form is usually fine.